src/main/java/picard/sam/BamErrorMetric/ReadBaseStratification.java

```diff
@@ -0,0 +1,916 @@
+     */
+    public static final RecordAndOffsetStratifier<ReadDirection> readDirectionStratifier = wrapStaticReadFunction(ReadDirection::of, "read_direction");
+
+    /**
+     * Stratifies bases into their read-pair's Orientation (i.e. F1R2, F2R1, F1F2 or R1R2)
+     */
+    public static final RecordAndOffsetStratifier<PairOrientation> readOrientationStratifier = wrapStaticReadFunction(PairOrientation::of, "pair_orientation");
+
+    /**
+     * Stratifies into quintiles of read cycle.
+     */
+    public static final BinnedReadCycleStratifier binnedReadCycleStratifier = new BinnedReadCycleStratifier();
+
+    /**
+     * Get the one-based cycle number of the base, taking the direction of the read into account
+     */
+    public static final RecordAndOffsetStratifier<Integer> baseCycleStratifier = wrapStaticFunction(ReadBaseStratification::stratifyCycle, "cycle");
+
+    /**
+     * Stratifies into the read-pairs estimated insert-length, as long as it isn't larger than 10x the length of the read
+     */
+    public static final RecordAndOffsetStratifier<Integer> insertLengthStratifier = wrapStaticReadFunction(ReadBaseStratification::stratifyInsertLength, "insert_length");
+
+    /**
+     * Stratifies into the base-quality of the base under consideration
+     */
+    public static final RecordAndOffsetStratifier<Byte> baseQualityStratifier = wrapStaticFunction(ReadBaseStratification::stratifyBaseQuality, "base_quality");
+
+    /**
+     * Stratifies into the mapping-quality of the read under consideration
+     */
+    public static final RecordAndOffsetStratifier<Integer> mappingQualityStratifier = wrapStaticReadFunction(ReadBaseStratification::stratifyMappingQuality, "mapping_quality");
+
+    /**
+     * Stratifies according to the overall mismatches (from NM) that the read has against the reference, NOT
+     * including the current base.
+     */
+    public static final MismatchesInReadStratifier mismatchesInReadStratifier = new MismatchesInReadStratifier();
+
+    /**
+     * Stratify by tags used during duplex and single index consensus calling.
+     */
+    public static final ConsensusStratifier consensusStratifier = new ConsensusStratifier();
+
+    /**
+     * Stratify by the number of Ns found in the read.
+     * This is particularly useful for data that has been consensus-called (a process that can add 'N' bases when there is no consensus)
+     */
+    public static final NsInReadStratifier nsInReadStratifier = new NsInReadStratifier();
+
+    /* *************** enums **************/
+
+    /**
+     * An Enum that is used to generate stratifiers from strings
+     * <p>
+     * To use this given a String 'str':
+     * <p>
+     * Stratifiers.valueOf(str).makeStratifier()
+     * <p>
+     * This is used in {@link CollectBamErrorMetrics} to convert an input argument to a fully functional {@link BaseErrorAggregation} object.
+     */
+    enum Stratifiers implements CommandLineParser.ClpEnum {
```

Comment:
In general enum names should be singular since it's not a collection of things, but a type. E.g. if you have one stratifier in your code you'd want declare it as `Stratifier s;` not `Stratifiers s;`